src/test/java/com/github/onsdigital/JWTVerifierImplTests.java

```diff
@@ -0,0 +1,111 @@
+    private final static String TOKEN_EXPIRED_TIME = "eyJhbGciOiJSUzI1NiIsImtpZCI6IjEyMzRleGFtcGxlPSJ9.eyJzdWIiOiJhYWFhYWFhYS1iYmJiLWNjY2MtZGRkZC1lZWVlZWVlZWVlZWUiLCJkZXZpY2Vfa2V5IjoiYWFhYWFhYWEtYmJiYi1jY2NjLWRkZGQtZWVlZWVlZWVlZWVlIiwiY29nbml0bzpncm91cHMiOlsiYWRtaW4iXSwidG9rZW5fdXNlIjoiYWNjZXNzIiwic2NvcGUiOiJhd3MuY29nbml0by5zaWduaW4udXNlci5hZG1pbiIsImF1dGhfdGltZSI6MTU2MjE5MDUyNCwiaXNzIjoiaHR0cHM6Ly9jb2duaXRvLWlkcC51cy13ZXN0LTIuYW1hem9uYXdzLmNvbS91cy13ZXN0LTJfZXhhbXBsZSIsImV4cCI6MTU2MjE5NDEyNCwiaWF0IjoxNTYyMTkwNTI0LCJvcmlnaW5fanRpIjoiYWFhYWFhYWEtYmJiYi1jY2NjLWRkZGQtZWVlZWVlZWVlZWVlIiwianRpIjoiYWFhYWFhYWEtYmJiYi1jY2NjLWRkZGQtZWVlZWVlZWVlZWVlIiwiY2xpZW50X2lkIjoiNTdjYmlzaGs0ajI0cGFiYzEyMzQ1Njc4OTAiLCJ1c2VybmFtZSI6ImphbmVkb2VAZXhhbXBsZS5jb20ifQ.Jo1Ex-iOlQw8BJaDUFP6rGJM4owdFs90ti53Upri_re01wvHssbDOmIKGjRPbHX8tPi-Gjpdsbdp00RmI56wiIZee7X6VvPjGOZAI-q4InNhS5JIKdJv_CPu33e5TTmOVjVvIOctm_uZNL7Yd9A0Haz3-3BzlT-nUJ5OgTIjYsecHlSpNg-ie7-YY34u40obMNvk4WpqU1zIdbHYpeB08gxHCSSnhXxu-4d1-XGJAWbINEyD93_steGsGDMS4VTiKQl96pAT7LnZLLiPjlkoAC-T9BsOQtK3riUhU2VM0KUTqyhM2rcgTQy-zs-E8_Yf37a77s1_FPkmNbYJsIDInQ";
+    private final static String PUBLIC_KEY = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAwdowpUQS8YPrzFamOi7bEM74+yeqoAH0Tm+PLZdWjaIxL13XfxX63jO3iHAJe/Uh74d2vtTuHV7Vkl0wRjjJOuWrj5espG5VxU6xHJG/R2tVDElv05fFZhZn/YCwUJV2Zrz9Z0PIdPHB4eZODo64Qfab+ihgeLlny5l6+srVacWqsQ47i3DdJQOd6MgacvrKt+B0Ior1HKlHR0NneFPNu9zvCg11O1W0yl2+ou0UOPVqF8EfoavvgwbAUWwxeHXcN+TmgKx6Oa4X9ylk+Q0FUewYF020f5j7AW9EZrdu4rBTO2fliV02+DblOh/NTuHEzeCBHp3h4Mg5Bv3NYxY9sQIDAQAB";
+    private final static String USERNAME = "janedoe@example.com";
+    private final static String USER_ID = "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee";
+    private final static String PUBLIC_KEY_ID = "1234example=";
+
+    /**
+     * Class under test
+     */
+    private JWTVerifierImpl verifier;
+
+    @BeforeEach
+    public void init() {
+        Map<String, String> signingKeys = new HashMap<>();
+        signingKeys.put(PUBLIC_KEY_ID, PUBLIC_KEY);
+        verifier = new JWTVerifierImpl(signingKeys);
+    }
+
+    @Test
+    void decodeAndVerifyValidJWTPayload() {
+        UserDataPayload jwtData = verifier.verify(SIGNED_TOKEN);
+
+        List<String> grps = jwtData.getGroups();
+
+        // assert json data
+        assertEquals(USER_ID, jwtData.getId());
+        assertEquals(USERNAME, jwtData.getEmail());
+        assertEquals("admin", grps.get(0));
```

Comment:
Consider asserting that the groups list is the expected length - this will help to catch future changes with unintended consequences.